controller/konnect/konnectextension_controller.go

```diff
@@ -326,82 +562,45 @@ func (r *KonnectExtensionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		}
 		ext.Status.DataPlaneClientAuth = &konnectv1alpha1.DataPlaneClientAuthStatus{
 			CertificateSecretRef: &konnectv1alpha1.SecretRef{
-				Name: secretRef.Name,
+				Name: certificateSecret.Name,
 			},
 		}
-		if err := r.Client.Status().Update(ctx, &ext); err != nil {
-			return ctrl.Result{}, err
-		}
+		updateExtensionStatus = true
 	}
 
-	if res, err := patch.StatusWithCondition(
-		ctx, r.Client, &ext,
-		kcfgconsts.ConditionType(konnectv1alpha1.KonnectExtensionReadyConditionType),
-		metav1.ConditionTrue,
-		kcfgconsts.ConditionReason(konnectv1alpha1.KonnectExtensionReadyReasonReady),
-		"KonnectExtension is ready",
-	); err != nil || !res.IsZero() {
-		return res, err
+	if ext.Status.DataPlaneClientAuth == nil ||
+		ext.Status.DataPlaneClientAuth.CertificateSecretRef == nil ||
+		ext.Status.DataPlaneClientAuth.CertificateSecretRef.Name != certificateSecret.Name {
+		ext.Status.DataPlaneClientAuth = &konnectv1alpha1.DataPlaneClientAuthStatus{
+			CertificateSecretRef: &konnectv1alpha1.SecretRef{
+				Name: certificateSecret.Name,
+			},
+		}
+		updateExtensionStatus = true
 	}
-
-	return ctrl.Result{}, nil
-}
-
-func getKonnectAPIAuthRefNN(ctx context.Context, cl client.Client, ext *konnectv1alpha1.KonnectExtension) (types.NamespacedName, error) {
-
-	if ext.Spec.KonnectConfiguration != nil {
-		// TODO: handle cross namespace refs when supported.
-		return types.NamespacedName{
-			Namespace: ext.Namespace,
-			Name:      ext.Spec.KonnectConfiguration.APIAuthConfigurationRef.Name,
-		}, nil
+	if updateExtensionStatus {
+		log.Debug(logger, "updating KonnectExtension status")
+		return ctrl.Result{}, r.Client.Status().Update(ctx, &ext)
```

Comment:
Could status update also return a conflict 🤔 ? If so I think we can use `k8serrors.IsConflict(err)` and requeue to prevent returning the error and to just requeue.

Reply:
yep, added an addional check